src/arch/arm/64/kernel/vspace.c

```diff
@@ -1404,13 +1422,17 @@ void deleteASID(asid_t asid, vspace_root_t *vspace)
 
     poolPtr = armKSASIDTable[asid >> asidLowBits];
 
-    if (poolPtr != NULL && poolPtr->array[asid & MASK(asidLowBits)] == vspace) {
-        invalidateTLBByASID(asid);
+    if (poolPtr != NULL) {
+        asid_map_t asid_map = poolPtr->array[asid & MASK(asidLowBits)];
+        if (asid_map_get_type(asid_map) == asid_map_asid_map_vspace &&
+            (vspace_root_t *)asid_map_asid_map_vspace_get_vspace_root(asid_map) == vspace) {
+            invalidateTLBByASID(asid);
 #ifdef CONFIG_ARM_HYPERVISOR_SUPPORT
-        invalidateASIDEntry(asid);
+            invalidateASIDEntry(asid);
```

Comment:
additional spaces?

Reply:
it's now under an additional `if` so the new indent looks correct to me